File: shoring/desk/src/main/java/com/panopset/marin/oldswpkg/games/blackjack/Layout.java
package com.panopset.marin.oldswpkg.games.blackjack;

import com.panopset.compat.Tile;
import com.panopset.blackjackEngine.CycleSnapshot;

public abstract class Layout {

    public final Tile t;
    public final int h;
    public final int w;
    public final int centerW;
    public final int centerH;
    public final int cardW;
    public final int cardH;
    public final int cardSpacer;
    public final CycleSnapshot cs;

    public Layout(Tile tile, int[] cardDim, CycleSnapshot cs) {
        t = tile;
        w = t.getWidth();
        h = t.getHeight();
        centerW = w / 2;
        centerH = h / 2;
        cardW = cardDim[0];
        cardH = cardDim[1];
        cardSpacer = cardW / 5;
        this.cs = cs;
    }
}

File: code/shoring/desk/src/main/kotlin/com/panopset/marin/oldswpkg/games/blackjack/Layout.kt
package com.panopset.marin.oldswpkg.games.blackjack

import com.panopset.blackjackEngine.CycleSnapshot
import com.panopset.compat.Tile

abstract class Layout(val t: Tile, cardDim: IntArray, val cs: CycleSnapshot) {
    val h: Int
    val w: Int
    val centerW: Int
    val centerH: Int
    val cardW: Int
    val cardH: Int
    val cardSpacer: Int

    init {
        w = t.width
        h = t.height
        centerW = w / 2
        centerH = h / 2
        cardW = cardDim[0]
        cardH = cardDim[1]
        cardSpacer = cardW / 5
    }
}
